tests/app/main/views/test_suppliers.py

```diff
@@ -1268,53 +1300,130 @@ def test_none_shown_if_user_not_ccs_admin(self, s3, get_signed_url, data_api_cli
         res = self.client.get('/admin/suppliers/1234/agreements/g-cloud-8')
         assert res.status_code == 200
 
-        data = res.get_data(as_text=True)
+        document = html.fromstring(res.get_data(as_text=True))
 
-        assert 'Accept and continue' not in data
-        assert 'Put on hold and continue' not in data
+        assert len(document.xpath("//input[@type='submit'][@value='Accept and continue']")) == 0
+        assert len(document.xpath("//input[@type='submit'][@value='Put on hold and continue']")) == 0
+        assert len(document.xpath("//h2[text()='Accepted by']")) == 0
+        assert len(document.xpath(
+            "//a[@href='/admin/suppliers/1234/agreements/g-cloud-8/next'][text()='Next agreement']"
+        )) == 1
 
     def test_both_shown_if_ccs_admin_and_agreement_signed(self, s3, get_signed_url, data_api_client):
         self.set_mocks(s3, get_signed_url, data_api_client, agreement_status='signed')
 
         res = self.client.get('/admin/suppliers/1234/agreements/g-cloud-8')
         assert res.status_code == 200
 
-        data = res.get_data(as_text=True)
-
-        assert 'Accept and continue' in data
-        assert 'Put on hold and continue' in data
+        document = html.fromstring(res.get_data(as_text=True))
+
+        assert len(document.xpath(
+            "//form[@action='/admin/suppliers/agreements/4321/approve']"
+            "//input[@type='submit'][@value='Accept and continue']"
+        )) == 1
+        assert len(document.xpath(
+            "//form[@action='/admin/suppliers/agreements/4321/on-hold']"
+            "//input[@type='submit'][@value='Put on hold and continue']"
+        )) == 1
+        assert len(document.xpath("//h2[text()='Accepted by']")) == 0
+        assert len(document.xpath(
+            "//a[@href='/admin/suppliers/1234/agreements/g-cloud-8/next'][text()='Next agreement']"
+        )) == 1
 
     def test_only_counter_sign_shown_if_agreement_on_hold(self, s3, get_signed_url, data_api_client):
         self.set_mocks(s3, get_signed_url, data_api_client, agreement_status='on-hold')
 
         res = self.client.get('/admin/suppliers/1234/agreements/g-cloud-8')
         assert res.status_code == 200
 
-        data = res.get_data(as_text=True)
+        document = html.fromstring(res.get_data(as_text=True))
 
-        assert 'Accept and continue' in data
-        assert 'Put on hold and continue' not in data
+        assert len(document.xpath(
+            "//form[@action='/admin/suppliers/agreements/4321/approve']"
+            "//input[@type='submit'][@value='Accept and continue']"
+        )) == 1
+        assert len(document.xpath("//input[@type='submit'][@value='Put on hold and continue']")) == 0
```

Comment:
Oh I see, because they're negative. Yeah. Hmm. I'm going to think about this. This one would fail to pick up if someone switched this to be a `<button>` and it started appearing.